generator/types.go

```diff
@@ -103,7 +121,7 @@ func NewMethod(name string, f *ast.FuncType, printer typePrinter) (*Method, erro
 }
 
 // NewParam returns Param struct
-func NewParam(name string, typ ast.Expr, usedNames map[string]bool, printer typePrinter) (*Param, error) {
+func NewParam(name string, typ ast.Expr, usedNames map[string]bool, printer typePrinter, fi *ast.Field) (*Param, error) {
```

Comment:
I think we can make "fi" a second parameter and get it's type inside the fuction rather than passing field and it's type separately. Look at line 169 for example.